

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="es" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="es" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Before we start &mdash; documentación de Call me by your reads - 1</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/language_data.js"></script>
        <script src="_static/clipboard.min.js"></script>
        <script src="_static/copybutton.js"></script>
        <script src="_static/translations.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/copybutton.css" type="text/css" />
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
    <link rel="index" title="Índice" href="genindex.html" />
    <link rel="search" title="Búsqueda" href="search.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> Call me by your reads
          

          
          </a>

          
            
            
              <div class="version">
                1
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <!-- Local TOC -->
              <div class="local-toc"><ul>
<li><a class="reference internal" href="#"><em>Before we start</em></a></li>
<li><a class="reference internal" href="#comandos-del-dia-4">Comandos del día 4</a></li>
<li><a class="reference internal" href="#comandos-del-dia-5">Comandos del día 5</a></li>
<li><a class="reference internal" href="#formatos-formatos-everywhere">Formatos… formatos everywhere</a></li>
</ul>
</div>
            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Call me by your reads</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li><em>Before we start</em></li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/gatk.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="before-we-start">
<h1><em>Before we start</em><a class="headerlink" href="#before-we-start" title="Enlazar permanentemente con este título">¶</a></h1>
<div class="admonition note">
<p class="admonition-title">Nota</p>
<p>En los ambientes unix linux los programas no se instalan de la misma forma que en los ambientes windows o mac, en vez tenemos que descargar el código fuente de los programas, compilarlo cuando así sea necesario e instalarlo en algun lugar de nuestro sistema</p>
<p>Los siguientes comandos nos permitirán instalar htslib, samtools y gatk en nuestra carpeta <code class="code docutils literal notranslate"><span class="pre">$HOME/bin</span></code></p>
</div>
<div class="toggle admonition">
<p class="admonition-title">htslib</p>
<blockquote>
<div><div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ pwd
/home/vflorelo

$ git clone https://github.com/samtools/htslib.git

$ cd /home/vflorelo/htslib

$ autoheader

$ autoconf

$ ./configure --prefix=/home/vflorelo

$ make

$ make install

$ cd /home/vflorelo
</pre></div>
</div>
</div></blockquote>
</div>
<div class="toggle admonition">
<p class="admonition-title">samtools</p>
<blockquote>
<div><div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ pwd
/home/vflorelo

$ git clone https://github.com/samtools/samtools.git

$ cd /home/vflorelo/samtools

$ autoheader

$ autoconf

$ ./configure --prefix=/home/vflorelo

$ make

$ make install

$ cd /home/vflorelo
</pre></div>
</div>
</div></blockquote>
</div>
<div class="toggle admonition">
<p class="admonition-title">GATK4</p>
<blockquote>
<div><div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ pwd
/home/vflorelo

$ wget https://github.com/broadinstitute/gatk/releases/download/4.1.8.1/gatk-4.1.8.1.zip

$ unzip gatk-4.1.8.1.zip

$ cd /home/vflorelo/gatk-4.1.8.1

$ mv * /home/vflorelo/bin

$ cd /home/vflorelo

$ chmod -R 775 /home/vflorelo/bin
</pre></div>
</div>
</div></blockquote>
</div>
</div>
<div class="section" id="comandos-del-dia-4">
<h1>Comandos del día 4<a class="headerlink" href="#comandos-del-dia-4" title="Enlazar permanentemente con este título">¶</a></h1>
<ol class="arabic" start="0">
<li><p>Preparamos el escenario, abramos nuestra terminal, nos vamos a la carpeta del día 4 y ponemos en ella el archivo en formato SAM que generamos el día 3, adicionalmente tendremos que copiar el archivo <code class="code docutils literal notranslate"><span class="pre">TruSight_One_xt_GRCh38.bed</span></code> que vive en la carpeta <code class="code docutils literal notranslate"><span class="pre">/usr/local/bioinformatics/databases</span></code></p>
<blockquote>
<div><div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ pwd
/home/vflorelo

$ mkdir dia_04

$ ls
dia_01 dia_02 dia_03 dia_04

$ mv dia_03/S1.sam dia_04/

$ cp /usr/local/bioinformatics/databases/TruSight_One_xt_GRCh38.bed dia_04/

$ cd dia_04

$ ls
S1.sam TruSight_One_xt_GRCh38.bed
</pre></div>
</div>
</div></blockquote>
</li>
<li><p>Primer paso: transformar nuestro archivo SAM a formato BAM, esencialmente son la misma cosa pero los archivos BAM están comprimidos y en binario</p>
<blockquote>
<div><div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">samtools</span> <span class="n">view</span> \
        <span class="o">-@</span> <span class="mi">4</span> \
        <span class="o">-</span><span class="n">b</span> \
        <span class="o">-</span><span class="n">h</span> \
        <span class="o">-</span><span class="n">o</span> <span class="n">S1</span><span class="o">.</span><span class="n">tmp</span><span class="o">.</span><span class="n">bam</span> \
        <span class="o">-</span><span class="n">f</span> <span class="mi">3</span> \
        <span class="o">-</span><span class="n">L</span> <span class="n">TruSight_One_xt_GRCh38</span><span class="o">.</span><span class="n">bed</span> <span class="n">S1</span><span class="o">.</span><span class="n">sam</span>
</pre></div>
</div>
<div class="admonition important">
<p class="admonition-title">Importante</p>
<p>Si todo sale bien, podemos eliminar nuestro archivo SAM para ahorrarnos espacio:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">rm</span> <span class="n">S1</span><span class="o">.</span><span class="n">sam</span>
</pre></div>
</div>
</div>
</div></blockquote>
</li>
<li><p>Posteriormente agregaremos un <a class="reference external" href="https://www.biostars.org/p/43897/">readgroup</a> a nuestro archivo BAM para indicarle</p></li>
</ol>
<ul>
<li><p>El identificador del <em>readgroup</em></p></li>
<li><p>El nombre de la biblioteca</p></li>
<li><p>La plataforma de secuenciación</p></li>
<li><p>El nombre de la muestra</p></li>
<li><p>La unidad de la plataforma (barcodes)</p>
<blockquote>
<div><div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">gatk</span> <span class="n">AddOrReplaceReadGroups</span> <span class="o">--</span><span class="n">INPUT</span> <span class="n">S1</span><span class="o">.</span><span class="n">tmp</span><span class="o">.</span><span class="n">bam</span> <span class="o">--</span><span class="n">OUTPUT</span> <span class="n">S1</span><span class="o">.</span><span class="n">rg</span><span class="o">.</span><span class="n">bam</span> <span class="o">--</span><span class="n">RGLB</span> <span class="n">S1</span> <span class="o">--</span><span class="n">RGID</span> <span class="n">S1</span> <span class="o">--</span><span class="n">RGPL</span> <span class="n">Illumina</span> <span class="o">--</span><span class="n">RGSM</span> <span class="n">S1</span> <span class="o">--</span><span class="n">RGPU</span> <span class="n">S1</span>
</pre></div>
</div>
</div></blockquote>
</li>
</ul>
<ol class="arabic" start="3">
<li><p>Hecho esto, ordenamos con base en las posiciones genómicas nuestro archivo BAM</p>
<blockquote>
<div><div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">gatk</span> <span class="n">SortSam</span> <span class="o">--</span><span class="n">INPUT</span> <span class="n">S1</span><span class="o">.</span><span class="n">rg</span><span class="o">.</span><span class="n">bam</span> <span class="o">--</span><span class="n">OUTPUT</span> <span class="n">S1</span><span class="o">.</span><span class="n">sorted</span><span class="o">.</span><span class="n">bam</span> <span class="o">--</span><span class="n">SORT_ORDER</span> <span class="n">coordinate</span>
</pre></div>
</div>
</div></blockquote>
</li>
<li><p>Paso importante (ó no). Marcado de duplicados</p>
<blockquote>
<div><div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">gatk</span> <span class="n">MarkDuplicates</span> <span class="o">--</span><span class="n">INPUT</span> <span class="n">S1</span><span class="o">.</span><span class="n">sorted</span><span class="o">.</span><span class="n">bam</span> <span class="o">--</span><span class="n">OUTPUT</span> <span class="n">S1</span><span class="o">.</span><span class="n">dupmarked</span><span class="o">.</span><span class="n">bam</span> <span class="o">--</span><span class="n">METRICS_FILE</span> <span class="n">S1</span><span class="o">.</span><span class="n">dupmarked</span><span class="o">.</span><span class="n">txt</span>
</pre></div>
</div>
</div></blockquote>
</li>
<li><p>Recalibración de los scores de calidad</p>
<blockquote>
<div><div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">gatk</span> <span class="n">BaseRecalibrator</span> \
        <span class="o">--</span><span class="n">reference</span> <span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">local</span><span class="o">/</span><span class="n">bioinformatics</span><span class="o">/</span><span class="n">databases</span><span class="o">/</span><span class="n">genome</span><span class="o">/</span><span class="n">Homo_sapiens_GRCh38</span><span class="o">.</span><span class="n">fasta</span> \
        <span class="o">--</span><span class="nb">input</span> <span class="n">S1</span><span class="o">.</span><span class="n">dupmarked</span><span class="o">.</span><span class="n">bam</span> \
        <span class="o">--</span><span class="n">known</span><span class="o">-</span><span class="n">sites</span> <span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">local</span><span class="o">/</span><span class="n">bioinformatics</span><span class="o">/</span><span class="n">databases</span><span class="o">/</span><span class="n">GATK4</span><span class="o">/</span><span class="n">dbsnp_146</span><span class="o">.</span><span class="n">hg38</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> \
        <span class="o">--</span><span class="n">known</span><span class="o">-</span><span class="n">sites</span> <span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">local</span><span class="o">/</span><span class="n">bioinformatics</span><span class="o">/</span><span class="n">databases</span><span class="o">/</span><span class="n">GATK4</span><span class="o">/</span><span class="n">Mills_and_1000G_gold_standard</span><span class="o">.</span><span class="n">indels</span><span class="o">.</span><span class="n">hg38</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> \
        <span class="o">--</span><span class="n">output</span> <span class="n">S1_recal</span><span class="o">-</span><span class="n">data</span><span class="o">.</span><span class="n">table</span> \
        <span class="o">--</span><span class="n">intervals</span> <span class="n">TruSight_One_xt_GRCh38</span><span class="o">.</span><span class="n">bed</span>
</pre></div>
</div>
</div></blockquote>
</li>
<li><p>Obtención de lecturas recalibradas</p>
<blockquote>
<div><div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">gatk</span> <span class="n">ApplyBQSR</span> \
        <span class="o">--</span><span class="n">reference</span> <span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">local</span><span class="o">/</span><span class="n">bioinformatics</span><span class="o">/</span><span class="n">databases</span><span class="o">/</span><span class="n">genome</span><span class="o">/</span><span class="n">Homo_sapiens_GRCh38</span><span class="o">.</span><span class="n">fasta</span> \
        <span class="o">--</span><span class="nb">input</span> <span class="n">S1</span><span class="o">.</span><span class="n">dupmarked</span><span class="o">.</span><span class="n">bam</span> \
        <span class="o">--</span><span class="n">bqsr</span><span class="o">-</span><span class="n">recal</span><span class="o">-</span><span class="n">file</span> <span class="n">S1_recal</span><span class="o">-</span><span class="n">data</span><span class="o">.</span><span class="n">table</span> \
        <span class="o">--</span><span class="n">output</span> <span class="n">S1_recal</span><span class="o">-</span><span class="n">reads</span><span class="o">.</span><span class="n">bam</span> \
        <span class="o">--</span><span class="n">intervals</span> <span class="n">TruSight_One_xt_GRCh38</span><span class="o">.</span><span class="n">bed</span>
</pre></div>
</div>
</div></blockquote>
</li>
<li><p>Obtención de scores de calidad de las lecturas recalibradas</p>
<blockquote>
<div><div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">gatk</span> <span class="n">BaseRecalibrator</span> \
        <span class="o">--</span><span class="n">reference</span> <span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">local</span><span class="o">/</span><span class="n">bioinformatics</span><span class="o">/</span><span class="n">databases</span><span class="o">/</span><span class="n">genome</span><span class="o">/</span><span class="n">Homo_sapiens_GRCh38</span><span class="o">.</span><span class="n">fasta</span> \
        <span class="o">--</span><span class="nb">input</span> <span class="n">S1_recal</span><span class="o">-</span><span class="n">reads</span><span class="o">.</span><span class="n">bam</span> \
        <span class="o">--</span><span class="n">known</span><span class="o">-</span><span class="n">sites</span> <span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">local</span><span class="o">/</span><span class="n">bioinformatics</span><span class="o">/</span><span class="n">databases</span><span class="o">/</span><span class="n">GATK4</span><span class="o">/</span><span class="n">dbsnp_146</span><span class="o">.</span><span class="n">hg38</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> \
        <span class="o">--</span><span class="n">known</span><span class="o">-</span><span class="n">sites</span> <span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">local</span><span class="o">/</span><span class="n">bioinformatics</span><span class="o">/</span><span class="n">databases</span><span class="o">/</span><span class="n">GATK4</span><span class="o">/</span><span class="n">Mills_and_1000G_gold_standard</span><span class="o">.</span><span class="n">indels</span><span class="o">.</span><span class="n">hg38</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> \
        <span class="o">--</span><span class="n">output</span> <span class="n">S1_post</span><span class="o">-</span><span class="n">recal</span><span class="o">-</span><span class="n">data</span><span class="o">.</span><span class="n">table</span> \
        <span class="o">--</span><span class="n">intervals</span> <span class="n">TruSight_One_xt_GRCh38</span><span class="o">.</span><span class="n">bed</span>
</pre></div>
</div>
</div></blockquote>
</li>
<li><p>Comparación de los scores de calidad</p>
<blockquote>
<div><div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">gatk</span> <span class="n">AnalyzeCovariates</span> \
        <span class="o">--</span><span class="n">before</span><span class="o">-</span><span class="n">report</span><span class="o">-</span><span class="n">file</span> <span class="n">S1_recal</span><span class="o">-</span><span class="n">data</span><span class="o">.</span><span class="n">table</span> \
        <span class="o">--</span><span class="n">after</span><span class="o">-</span><span class="n">report</span><span class="o">-</span><span class="n">file</span> <span class="n">S1_post</span><span class="o">-</span><span class="n">recal</span><span class="o">-</span><span class="n">data</span><span class="o">.</span><span class="n">table</span> \
        <span class="o">--</span><span class="n">plots</span><span class="o">-</span><span class="n">report</span><span class="o">-</span><span class="n">file</span> <span class="n">S1_recal</span><span class="o">-</span><span class="n">plots</span><span class="o">.</span><span class="n">pdf</span>
</pre></div>
</div>
</div></blockquote>
</li>
</ol>
</div>
<div class="section" id="comandos-del-dia-5">
<h1>Comandos del día 5<a class="headerlink" href="#comandos-del-dia-5" title="Enlazar permanentemente con este título">¶</a></h1>
<ol class="arabic" start="9">
<li><p>Llamado de variantes <strong>crudas</strong></p>
<blockquote>
<div><p>9.1 Llamado de variantes de línea germinal por individuo en modo <em>single-sample</em></p>
<blockquote>
<div><div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">gatk</span> <span class="n">HaplotypeCaller</span> \
        <span class="o">--</span><span class="n">reference</span> <span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">local</span><span class="o">/</span><span class="n">bioinformatics</span><span class="o">/</span><span class="n">databases</span><span class="o">/</span><span class="n">genome</span><span class="o">/</span><span class="n">Homo_sapiens_GRCh38</span><span class="o">.</span><span class="n">fasta</span> \
        <span class="o">--</span><span class="nb">input</span> <span class="n">S1_recal</span><span class="o">-</span><span class="n">reads</span><span class="o">.</span><span class="n">bam</span> \
        <span class="o">--</span><span class="n">intervals</span> <span class="n">TruSight_One_xt_GRCh38</span><span class="o">.</span><span class="n">bed</span> \
        <span class="o">--</span><span class="n">stand</span><span class="o">-</span><span class="n">call</span><span class="o">-</span><span class="n">conf</span> <span class="mf">10.0</span> \
        <span class="o">--</span><span class="n">output</span> <span class="n">S1_raw</span><span class="o">-</span><span class="nb">vars</span><span class="o">.</span><span class="n">vcf</span>        \
        <span class="o">--</span><span class="n">native</span><span class="o">-</span><span class="n">pair</span><span class="o">-</span><span class="n">hmm</span><span class="o">-</span><span class="n">threads</span> <span class="mi">4</span>
</pre></div>
</div>
</div></blockquote>
<p>9.2 Llamado de variantes de línea germinal por individuo para joint-genotype</p>
<blockquote>
<div><div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">gatk</span> <span class="n">HaplotypeCaller</span> \
        <span class="o">--</span><span class="n">reference</span> <span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">local</span><span class="o">/</span><span class="n">bioinformatics</span><span class="o">/</span><span class="n">databases</span><span class="o">/</span><span class="n">genome</span><span class="o">/</span><span class="n">Homo_sapiens_GRCh38</span><span class="o">.</span><span class="n">fasta</span> \
        <span class="o">--</span><span class="nb">input</span> <span class="n">S1_recal</span><span class="o">-</span><span class="n">reads</span><span class="o">.</span><span class="n">bam</span> \
        <span class="o">--</span><span class="n">intervals</span> <span class="n">TruSight_One_xt_GRCh38</span><span class="o">.</span><span class="n">bed</span> \
        <span class="o">--</span><span class="n">stand</span><span class="o">-</span><span class="n">call</span><span class="o">-</span><span class="n">conf</span> <span class="mf">10.0</span> \
        <span class="o">--</span><span class="n">output</span> <span class="n">S1_raw</span><span class="o">-</span><span class="nb">vars</span><span class="o">.</span><span class="n">g</span><span class="o">.</span><span class="n">vcf</span> \
        <span class="o">--</span><span class="n">native</span><span class="o">-</span><span class="n">pair</span><span class="o">-</span><span class="n">hmm</span><span class="o">-</span><span class="n">threads</span> <span class="mi">4</span> \
        <span class="o">--</span><span class="n">emit</span><span class="o">-</span><span class="n">ref</span><span class="o">-</span><span class="n">confidence</span> <span class="n">GVCF</span> \
        <span class="o">--</span><span class="n">output</span><span class="o">-</span><span class="n">mode</span> <span class="n">EMIT_ALL_CONFIDENT_SITES</span>

<span class="n">gatk</span> <span class="n">HaplotypeCaller</span> \
        <span class="o">--</span><span class="n">reference</span> <span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">local</span><span class="o">/</span><span class="n">bioinformatics</span><span class="o">/</span><span class="n">databases</span><span class="o">/</span><span class="n">genome</span><span class="o">/</span><span class="n">Homo_sapiens_GRCh38</span><span class="o">.</span><span class="n">fasta</span> \
        <span class="o">--</span><span class="nb">input</span> <span class="n">S2_recal</span><span class="o">-</span><span class="n">reads</span><span class="o">.</span><span class="n">bam</span> \
        <span class="o">--</span><span class="n">intervals</span> <span class="n">TruSight_One_xt_GRCh38</span><span class="o">.</span><span class="n">bed</span> \
        <span class="o">--</span><span class="n">stand</span><span class="o">-</span><span class="n">call</span><span class="o">-</span><span class="n">conf</span> <span class="mf">10.0</span> \
        <span class="o">--</span><span class="n">output</span> <span class="n">S2_raw</span><span class="o">-</span><span class="nb">vars</span><span class="o">.</span><span class="n">g</span><span class="o">.</span><span class="n">vcf</span> \
        <span class="o">--</span><span class="n">native</span><span class="o">-</span><span class="n">pair</span><span class="o">-</span><span class="n">hmm</span><span class="o">-</span><span class="n">threads</span> <span class="mi">4</span> \
        <span class="o">--</span><span class="n">emit</span><span class="o">-</span><span class="n">ref</span><span class="o">-</span><span class="n">confidence</span> <span class="n">GVCF</span> \
        <span class="o">--</span><span class="n">output</span><span class="o">-</span><span class="n">mode</span> <span class="n">EMIT_ALL_CONFIDENT_SITES</span>

<span class="n">gatk</span> <span class="n">HaplotypeCaller</span> \
        <span class="o">--</span><span class="n">reference</span> <span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">local</span><span class="o">/</span><span class="n">bioinformatics</span><span class="o">/</span><span class="n">databases</span><span class="o">/</span><span class="n">genome</span><span class="o">/</span><span class="n">Homo_sapiens_GRCh38</span><span class="o">.</span><span class="n">fasta</span> \
        <span class="o">--</span><span class="nb">input</span> <span class="n">S3_recal</span><span class="o">-</span><span class="n">reads</span><span class="o">.</span><span class="n">bam</span> \
        <span class="o">--</span><span class="n">intervals</span> <span class="n">TruSight_One_xt_GRCh38</span><span class="o">.</span><span class="n">bed</span> \
        <span class="o">--</span><span class="n">stand</span><span class="o">-</span><span class="n">call</span><span class="o">-</span><span class="n">conf</span> <span class="mf">10.0</span> \
        <span class="o">--</span><span class="n">output</span> <span class="n">S3_raw</span><span class="o">-</span><span class="nb">vars</span><span class="o">.</span><span class="n">g</span><span class="o">.</span><span class="n">vcf</span> \
        <span class="o">--</span><span class="n">native</span><span class="o">-</span><span class="n">pair</span><span class="o">-</span><span class="n">hmm</span><span class="o">-</span><span class="n">threads</span> <span class="mi">4</span> \
        <span class="o">--</span><span class="n">emit</span><span class="o">-</span><span class="n">ref</span><span class="o">-</span><span class="n">confidence</span> <span class="n">GVCF</span> \
        <span class="o">--</span><span class="n">output</span><span class="o">-</span><span class="n">mode</span> <span class="n">EMIT_ALL_CONFIDENT_SITES</span>

<span class="n">gatk</span> <span class="n">GenomicsDBImport</span> \
        <span class="o">--</span><span class="n">variant</span> <span class="n">S1</span><span class="o">.</span><span class="n">g</span><span class="o">.</span><span class="n">vcf</span> \
        <span class="o">--</span><span class="n">variant</span> <span class="n">S2</span><span class="o">.</span><span class="n">g</span><span class="o">.</span><span class="n">vcf</span> \
        <span class="o">--</span><span class="n">variant</span> <span class="n">S3</span><span class="o">.</span><span class="n">g</span><span class="o">.</span><span class="n">vcf</span> \
        <span class="o">--</span><span class="n">genomicsdb</span><span class="o">-</span><span class="n">workspace</span><span class="o">-</span><span class="n">path</span> <span class="n">my_database</span> \
        <span class="o">--</span><span class="n">intervals</span> <span class="mi">1</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">3</span><span class="p">,</span><span class="mi">4</span><span class="p">,</span><span class="mi">5</span><span class="p">,</span><span class="mi">6</span><span class="p">,</span><span class="mi">7</span><span class="p">,</span><span class="mi">8</span><span class="p">,</span><span class="mi">9</span><span class="p">,</span><span class="mi">10</span><span class="p">,</span><span class="mi">11</span><span class="p">,</span><span class="mi">12</span><span class="p">,</span><span class="mi">13</span><span class="p">,</span><span class="mi">14</span><span class="p">,</span><span class="mi">15</span><span class="p">,</span><span class="mi">16</span><span class="p">,</span><span class="mi">17</span><span class="p">,</span><span class="mi">18</span><span class="p">,</span><span class="mi">19</span><span class="p">,</span><span class="mi">20</span><span class="p">,</span><span class="mi">21</span><span class="p">,</span><span class="mi">22</span><span class="p">,</span><span class="n">X</span><span class="p">,</span><span class="n">Y</span><span class="p">,</span><span class="n">MT</span>

<span class="n">gatk</span> <span class="n">GenotypeGVCFs</span> \
        <span class="o">--</span><span class="n">reference</span> <span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">local</span><span class="o">/</span><span class="n">bioinformatics</span><span class="o">/</span><span class="n">databases</span><span class="o">/</span><span class="n">genome</span><span class="o">/</span><span class="n">Homo_sapiens_GRCh38</span><span class="o">.</span><span class="n">fasta</span> \
        <span class="o">--</span><span class="n">variant</span> <span class="n">gendb</span><span class="p">:</span><span class="o">//</span><span class="n">my_database</span> \
        <span class="o">--</span><span class="n">use</span><span class="o">-</span><span class="n">new</span><span class="o">-</span><span class="n">qual</span><span class="o">-</span><span class="n">calculator</span> \
        <span class="o">--</span><span class="n">output</span> <span class="n">joint_raw</span><span class="o">-</span><span class="nb">vars</span><span class="o">.</span><span class="n">vcf</span>
</pre></div>
</div>
</div></blockquote>
<p>9.3 Llamado de variantes somáticas por individuo en modo tumor-control</p>
<blockquote>
<div><div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">gatk</span> <span class="n">Mutect2</span> \
        <span class="o">--</span><span class="n">reference</span> <span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">local</span><span class="o">/</span><span class="n">bioinformatics</span><span class="o">/</span><span class="n">databases</span><span class="o">/</span><span class="n">genome</span><span class="o">/</span><span class="n">Homo_sapiens_GRCh38</span><span class="o">.</span><span class="n">fasta</span> \
        <span class="o">--</span><span class="nb">input</span> <span class="n">S2_recal</span><span class="o">-</span><span class="n">reads</span><span class="o">.</span><span class="n">bam</span> \
        <span class="o">--</span><span class="n">tumor</span><span class="o">-</span><span class="n">sample</span> <span class="n">S2</span> \
        <span class="o">--</span><span class="nb">input</span> <span class="n">S1_recal</span><span class="o">-</span><span class="n">reads</span><span class="o">.</span><span class="n">bam</span> \
        <span class="o">--</span><span class="n">normal</span><span class="o">-</span><span class="n">sample</span> <span class="n">S1</span> \
        <span class="o">--</span><span class="n">germline</span><span class="o">-</span><span class="n">resource</span> <span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">local</span><span class="o">/</span><span class="n">bioinformatics</span><span class="o">/</span><span class="n">databases</span><span class="o">/</span><span class="n">af</span><span class="o">-</span><span class="n">only</span><span class="o">-</span><span class="n">gnomad</span><span class="o">.</span><span class="n">hg38</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> \
        <span class="o">--</span><span class="n">panel</span><span class="o">-</span><span class="n">of</span><span class="o">-</span><span class="n">normals</span> <span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">local</span><span class="o">/</span><span class="n">bioinformatics</span><span class="o">/</span><span class="n">databases</span><span class="o">/</span><span class="mi">1000</span><span class="n">g_pon</span><span class="o">.</span><span class="n">hg38</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> \
        <span class="o">--</span><span class="n">intervals</span> <span class="n">ClearSeq_Comprehensive_Cancer_GRCh38</span><span class="o">.</span><span class="n">bed</span> \
        <span class="o">--</span><span class="n">output</span> <span class="n">S2_S1</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span>
</pre></div>
</div>
</div></blockquote>
<p>9.4 Llamado de variantes somáticas por individuo en modo tumor-only</p>
<blockquote>
<div><div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">gatk</span> <span class="n">Mutect2</span> \
<span class="o">--</span><span class="n">reference</span> <span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">local</span><span class="o">/</span><span class="n">bioinformatics</span><span class="o">/</span><span class="n">databases</span><span class="o">/</span><span class="n">genome</span><span class="o">/</span><span class="n">Homo_sapiens_GRCh38</span><span class="o">.</span><span class="n">fasta</span> \
<span class="o">--</span><span class="nb">input</span> <span class="n">S2_recal</span><span class="o">-</span><span class="n">reads</span><span class="o">.</span><span class="n">bam</span> \
<span class="o">--</span><span class="n">tumor</span><span class="o">-</span><span class="n">sample</span> <span class="n">S2</span> \
<span class="o">--</span><span class="n">germline</span><span class="o">-</span><span class="n">resource</span> <span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">local</span><span class="o">/</span><span class="n">bioinformatics</span><span class="o">/</span><span class="n">databases</span><span class="o">/</span><span class="n">af</span><span class="o">-</span><span class="n">only</span><span class="o">-</span><span class="n">gnomad</span><span class="o">.</span><span class="n">hg38</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> \
<span class="o">--</span><span class="n">panel</span><span class="o">-</span><span class="n">of</span><span class="o">-</span><span class="n">normals</span> <span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">local</span><span class="o">/</span><span class="n">bioinformatics</span><span class="o">/</span><span class="n">databases</span><span class="o">/</span><span class="mi">1000</span><span class="n">g_pon</span><span class="o">.</span><span class="n">hg38</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> \
<span class="o">--</span><span class="n">intervals</span> <span class="n">ClearSeq_Comprehensive_Cancer_GRCh38</span><span class="o">.</span><span class="n">bed</span> \
<span class="o">--</span><span class="n">output</span> <span class="n">S2</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span>
</pre></div>
</div>
</div></blockquote>
</div></blockquote>
</li>
<li><p>Recalibración de SNVs</p>
<blockquote>
<div><div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">gatk</span> <span class="n">VariantRecalibrator</span> \
        <span class="o">--</span><span class="n">reference</span> <span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">local</span><span class="o">/</span><span class="n">bioinformatics</span><span class="o">/</span><span class="n">databases</span><span class="o">/</span><span class="n">Genome</span><span class="o">/</span><span class="n">Homo_sapiens_GRCh38</span><span class="o">.</span><span class="n">fasta</span> \
        <span class="o">--</span><span class="n">variant</span> <span class="n">S1_raw</span><span class="o">-</span><span class="nb">vars</span><span class="o">.</span><span class="n">vcf</span> \
        <span class="o">--</span><span class="n">intervals</span> <span class="n">TruSight_One_xt_GRCh38</span><span class="o">.</span><span class="n">bed</span> \
        <span class="o">--</span><span class="n">resource</span><span class="p">:</span><span class="n">hapmap</span><span class="p">,</span><span class="n">known</span><span class="o">=</span><span class="n">false</span><span class="p">,</span><span class="n">training</span><span class="o">=</span><span class="n">true</span><span class="p">,</span><span class="n">truth</span><span class="o">=</span><span class="n">true</span><span class="p">,</span><span class="n">prior</span><span class="o">=</span><span class="mf">15.0</span> <span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">local</span><span class="o">/</span><span class="n">bioinformatics</span><span class="o">/</span><span class="n">databases</span><span class="o">/</span><span class="n">GATK4</span><span class="o">/</span><span class="n">hapmap_3</span><span class="o">.</span><span class="mf">3.</span><span class="n">hg38</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> \
        <span class="o">--</span><span class="n">resource</span><span class="p">:</span><span class="n">omni</span><span class="p">,</span><span class="n">known</span><span class="o">=</span><span class="n">false</span><span class="p">,</span><span class="n">training</span><span class="o">=</span><span class="n">true</span><span class="p">,</span><span class="n">truth</span><span class="o">=</span><span class="n">true</span><span class="p">,</span><span class="n">prior</span><span class="o">=</span><span class="mf">12.0</span> <span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">local</span><span class="o">/</span><span class="n">bioinformatics</span><span class="o">/</span><span class="n">databases</span><span class="o">/</span><span class="n">GATK4</span><span class="o">/</span><span class="mi">1000</span><span class="n">G_omni2</span><span class="o">.</span><span class="mf">5.</span><span class="n">hg38</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> \
        <span class="o">--</span><span class="n">resource</span><span class="p">:</span><span class="mi">1000</span><span class="n">G</span><span class="p">,</span><span class="n">known</span><span class="o">=</span><span class="n">false</span><span class="p">,</span><span class="n">training</span><span class="o">=</span><span class="n">true</span><span class="p">,</span><span class="n">truth</span><span class="o">=</span><span class="n">false</span><span class="p">,</span><span class="n">prior</span><span class="o">=</span><span class="mf">10.0</span> <span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">local</span><span class="o">/</span><span class="n">bioinformatics</span><span class="o">/</span><span class="n">databases</span><span class="o">/</span><span class="n">GATK4</span><span class="o">/</span><span class="mi">1000</span><span class="n">G_phase1</span><span class="o">.</span><span class="n">snps</span><span class="o">.</span><span class="n">high_confidence</span><span class="o">.</span><span class="n">hg38</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> \
        <span class="o">--</span><span class="n">resource</span><span class="p">:</span><span class="n">dbsnp</span><span class="p">,</span><span class="n">known</span><span class="o">=</span><span class="n">true</span><span class="p">,</span><span class="n">training</span><span class="o">=</span><span class="n">false</span><span class="p">,</span><span class="n">truth</span><span class="o">=</span><span class="n">false</span><span class="p">,</span><span class="n">prior</span><span class="o">=</span><span class="mf">2.0</span> <span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">local</span><span class="o">/</span><span class="n">bioinformatics</span><span class="o">/</span><span class="n">databases</span><span class="o">/</span><span class="n">GATK4</span><span class="o">/</span><span class="n">dbsnp_146</span><span class="o">.</span><span class="n">hg38</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> \
        <span class="o">--</span><span class="n">use</span><span class="o">-</span><span class="n">annotation</span> <span class="n">QD</span> \
        <span class="o">--</span><span class="n">use</span><span class="o">-</span><span class="n">annotation</span> <span class="n">FS</span> \
        <span class="o">--</span><span class="n">use</span><span class="o">-</span><span class="n">annotation</span> <span class="n">SOR</span> \
        <span class="o">--</span><span class="n">use</span><span class="o">-</span><span class="n">annotation</span> <span class="n">MQ</span> \
        <span class="o">--</span><span class="n">use</span><span class="o">-</span><span class="n">annotation</span> <span class="n">MQRankSum</span> \
        <span class="o">--</span><span class="n">use</span><span class="o">-</span><span class="n">annotation</span> <span class="n">ReadPosRankSum</span> \
        <span class="o">--</span><span class="n">mode</span> <span class="n">SNP</span> \
        <span class="o">--</span><span class="n">truth</span><span class="o">-</span><span class="n">sensitivity</span><span class="o">-</span><span class="n">tranche</span> <span class="mf">100.0</span> \
        <span class="o">--</span><span class="n">truth</span><span class="o">-</span><span class="n">sensitivity</span><span class="o">-</span><span class="n">tranche</span> <span class="mf">99.9</span> \
        <span class="o">--</span><span class="n">truth</span><span class="o">-</span><span class="n">sensitivity</span><span class="o">-</span><span class="n">tranche</span> <span class="mf">99.0</span> \
        <span class="o">--</span><span class="n">truth</span><span class="o">-</span><span class="n">sensitivity</span><span class="o">-</span><span class="n">tranche</span> <span class="mf">90.0</span> \
        <span class="o">--</span><span class="nb">max</span><span class="o">-</span><span class="n">gaussians</span> <span class="mi">1</span> \
        <span class="o">--</span><span class="nb">max</span><span class="o">-</span><span class="n">negative</span><span class="o">-</span><span class="n">gaussians</span> <span class="mi">1</span> \
        <span class="o">--</span><span class="n">output</span> <span class="n">S1_recalibrate</span><span class="o">-</span><span class="n">SNP</span><span class="o">.</span><span class="n">recal</span> \
        <span class="o">--</span><span class="n">tranches</span><span class="o">-</span><span class="n">file</span> <span class="n">S1_recalibrate</span><span class="o">-</span><span class="n">SNP</span><span class="o">.</span><span class="n">tranches</span> \
        <span class="o">--</span><span class="n">rscript</span><span class="o">-</span><span class="n">file</span> <span class="n">S1_recalibrate</span><span class="o">-</span><span class="n">SNP</span><span class="o">-</span><span class="n">plots</span><span class="o">.</span><span class="n">R</span>
<span class="n">gatk</span> <span class="n">ApplyVQSR</span> \
        <span class="o">--</span><span class="n">reference</span> <span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">local</span><span class="o">/</span><span class="n">bioinformatics</span><span class="o">/</span><span class="n">databases</span><span class="o">/</span><span class="n">Genome</span><span class="o">/</span><span class="n">Homo_sapiens_GRCh38</span><span class="o">.</span><span class="n">fasta</span> \
        <span class="o">--</span><span class="n">variant</span> <span class="n">S1_raw</span><span class="o">-</span><span class="nb">vars</span><span class="o">.</span><span class="n">vcf</span> \
        <span class="o">--</span><span class="n">intervals</span> <span class="n">TruSight_One_xt_GRCh38</span><span class="o">.</span><span class="n">bed</span> \
        <span class="o">--</span><span class="n">mode</span> <span class="n">SNP</span> \
        <span class="o">--</span><span class="n">truth</span><span class="o">-</span><span class="n">sensitivity</span><span class="o">-</span><span class="nb">filter</span><span class="o">-</span><span class="n">level</span> <span class="mf">99.0</span> \
        <span class="o">--</span><span class="n">recal</span><span class="o">-</span><span class="n">file</span> <span class="n">S1_recalibrate</span><span class="o">-</span><span class="n">SNP</span><span class="o">.</span><span class="n">recal</span> \
        <span class="o">--</span><span class="n">tranches</span><span class="o">-</span><span class="n">file</span> <span class="n">S1_recalibrate</span><span class="o">-</span><span class="n">SNP</span><span class="o">.</span><span class="n">tranches</span> \
        <span class="o">--</span><span class="n">output</span> <span class="n">S1_recal</span><span class="o">-</span><span class="n">snps_raw</span><span class="o">-</span><span class="n">indels</span><span class="o">.</span><span class="n">vcf</span>
</pre></div>
</div>
</div></blockquote>
</li>
<li><p>Recalibración de InDels</p>
<blockquote>
<div><div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">gatk</span> <span class="n">VariantRecalibrator</span> \
        <span class="o">--</span><span class="n">reference</span> <span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">local</span><span class="o">/</span><span class="n">bioinformatics</span><span class="o">/</span><span class="n">databases</span><span class="o">/</span><span class="n">Genome</span><span class="o">/</span><span class="n">Homo_sapiens_GRCh38</span><span class="o">.</span><span class="n">fasta</span> \
        <span class="o">--</span><span class="n">variant</span> <span class="n">S1_recal</span><span class="o">-</span><span class="n">snps_raw</span><span class="o">-</span><span class="n">indels</span><span class="o">.</span><span class="n">vcf</span> \
        <span class="o">--</span><span class="n">intervals</span> <span class="n">TruSight_One_xt_GRCh38</span><span class="o">.</span><span class="n">bed</span> \
        <span class="o">--</span><span class="n">resource</span><span class="p">:</span><span class="n">mills</span><span class="p">,</span><span class="n">known</span><span class="o">=</span><span class="n">true</span><span class="p">,</span><span class="n">training</span><span class="o">=</span><span class="n">true</span><span class="p">,</span><span class="n">truth</span><span class="o">=</span><span class="n">true</span><span class="p">,</span><span class="n">prior</span><span class="o">=</span><span class="mf">12.0</span> <span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">local</span><span class="o">/</span><span class="n">bioinformatics</span><span class="o">/</span><span class="n">databases</span><span class="o">/</span><span class="n">GATK4</span><span class="o">/</span><span class="n">Mills_and_1000G_gold_standard</span><span class="o">.</span><span class="n">indels</span><span class="o">.</span><span class="n">hg38</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> \
        <span class="o">--</span><span class="n">use</span><span class="o">-</span><span class="n">annotation</span> <span class="n">QD</span> \
        <span class="o">--</span><span class="n">use</span><span class="o">-</span><span class="n">annotation</span> <span class="n">FS</span> \
        <span class="o">--</span><span class="n">use</span><span class="o">-</span><span class="n">annotation</span> <span class="n">SOR</span> \
        <span class="o">--</span><span class="n">use</span><span class="o">-</span><span class="n">annotation</span> <span class="n">MQRankSum</span> \
        <span class="o">--</span><span class="n">use</span><span class="o">-</span><span class="n">annotation</span> <span class="n">ReadPosRankSum</span> \
        <span class="o">--</span><span class="n">mode</span> <span class="n">INDEL</span> \
        <span class="o">--</span><span class="n">truth</span><span class="o">-</span><span class="n">sensitivity</span><span class="o">-</span><span class="n">tranche</span> <span class="mf">100.0</span> \
        <span class="o">--</span><span class="n">truth</span><span class="o">-</span><span class="n">sensitivity</span><span class="o">-</span><span class="n">tranche</span> <span class="mf">99.9</span> \
        <span class="o">--</span><span class="n">truth</span><span class="o">-</span><span class="n">sensitivity</span><span class="o">-</span><span class="n">tranche</span> <span class="mf">99.0</span> \
        <span class="o">--</span><span class="n">truth</span><span class="o">-</span><span class="n">sensitivity</span><span class="o">-</span><span class="n">tranche</span> <span class="mf">90.0</span> \
        <span class="o">--</span><span class="nb">max</span><span class="o">-</span><span class="n">gaussians</span> <span class="mi">1</span> \
        <span class="o">--</span><span class="nb">max</span><span class="o">-</span><span class="n">negative</span><span class="o">-</span><span class="n">gaussians</span> <span class="mi">1</span> \
        <span class="o">--</span><span class="n">output</span> <span class="n">S1_recalibrate</span><span class="o">-</span><span class="n">INDEL</span><span class="o">.</span><span class="n">recal</span> \
        <span class="o">--</span><span class="n">tranches</span><span class="o">-</span><span class="n">file</span> <span class="n">S1_recalibrate</span><span class="o">-</span><span class="n">INDEL</span><span class="o">.</span><span class="n">tranches</span> \
        <span class="o">--</span><span class="n">rscript</span><span class="o">-</span><span class="n">file</span> <span class="n">S1_recalibrate</span><span class="o">-</span><span class="n">INDEL</span><span class="o">-</span><span class="n">plots</span><span class="o">.</span><span class="n">R</span>
<span class="n">gatk</span> <span class="n">ApplyVQSR</span> \
        <span class="o">--</span><span class="n">reference</span> <span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">local</span><span class="o">/</span><span class="n">bioinformatics</span><span class="o">/</span><span class="n">databases</span><span class="o">/</span><span class="n">Genome</span><span class="o">/</span><span class="n">Homo_sapiens_GRCh38</span><span class="o">.</span><span class="n">fasta</span> \
        <span class="o">--</span><span class="n">variant</span> <span class="n">S1_recal</span><span class="o">-</span><span class="n">snps_raw</span><span class="o">-</span><span class="n">indels</span><span class="o">.</span><span class="n">vcf</span> \
        <span class="o">--</span><span class="n">intervals</span> <span class="n">TruSight_One_xt_GRCh38</span><span class="o">.</span><span class="n">bed</span> \
        <span class="o">--</span><span class="n">mode</span> <span class="n">INDEL</span> \
        <span class="o">--</span><span class="n">truth</span><span class="o">-</span><span class="n">sensitivity</span><span class="o">-</span><span class="nb">filter</span><span class="o">-</span><span class="n">level</span> <span class="mf">99.0</span> \
        <span class="o">--</span><span class="n">recal</span><span class="o">-</span><span class="n">file</span> <span class="n">S1_recalibrate</span><span class="o">-</span><span class="n">INDEL</span><span class="o">.</span><span class="n">recal</span> \
        <span class="o">--</span><span class="n">tranches</span><span class="o">-</span><span class="n">file</span> <span class="n">S1_recalibrate</span><span class="o">-</span><span class="n">INDEL</span><span class="o">.</span><span class="n">tranches</span> \
        <span class="o">--</span><span class="n">output</span> <span class="n">S1_recalibrated_variants</span><span class="o">.</span><span class="n">vcf</span>
</pre></div>
</div>
</div></blockquote>
</li>
<li><p>Anotación de variantes: ID</p>
<blockquote>
<div><div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">gatk</span> <span class="n">VariantAnnotator</span> \
        <span class="o">--</span><span class="n">reference</span> <span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">local</span><span class="o">/</span><span class="n">bioinformatics</span><span class="o">/</span><span class="n">databases</span><span class="o">/</span><span class="n">Genome</span><span class="o">/</span><span class="n">Homo_sapiens_GRCh38</span><span class="o">.</span><span class="n">fasta</span> \
        <span class="o">--</span><span class="n">variant</span> <span class="n">S1_recalibrated_variants</span><span class="o">.</span><span class="n">vcf</span> \
        <span class="o">--</span><span class="n">intervals</span> <span class="n">TruSight_One_xt_GRCh38</span><span class="o">.</span><span class="n">bed</span> \
        <span class="o">--</span><span class="n">dbsnp</span> <span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">local</span><span class="o">/</span><span class="n">bioinformatics</span><span class="o">/</span><span class="n">databases</span><span class="o">/</span><span class="n">dbSNP</span><span class="o">/</span><span class="n">dbSNP</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> \
        <span class="o">--</span><span class="n">output</span> <span class="n">S1_annotated_variants</span><span class="o">.</span><span class="n">vcf</span> \
        <span class="o">--</span><span class="n">annotation</span> <span class="n">Coverage</span>
</pre></div>
</div>
</div></blockquote>
</li>
<li><p>Selección de variantes</p>
<blockquote>
<div><div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">gatk</span> <span class="n">SelectVariants</span> \
        <span class="o">--</span><span class="n">reference</span> <span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">local</span><span class="o">/</span><span class="n">bioinformatics</span><span class="o">/</span><span class="n">databases</span><span class="o">/</span><span class="n">Genome</span><span class="o">/</span><span class="n">Homo_sapiens_GRCh38</span><span class="o">.</span><span class="n">fasta</span> \
        <span class="o">--</span><span class="n">variant</span> <span class="n">S1_annotated_variants</span><span class="o">.</span><span class="n">vcf</span> \
        <span class="o">--</span><span class="n">output</span> <span class="n">S1_annotated_qd_dp_filtered_variants</span><span class="o">.</span><span class="n">vcf</span> \
        <span class="o">--</span><span class="n">selectExpressions</span> <span class="s2">&quot;QD &gt; 5.0 &amp;&amp; DP &gt; 10.0&quot;</span>
</pre></div>
</div>
</div></blockquote>
</li>
<li><p>Anotación funcional de variantes</p>
<blockquote>
<div><div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">gatk</span> <span class="n">Funcotator</span> \
        <span class="o">--</span><span class="n">variant</span> <span class="n">S1_annotated_qd_dp_filtered_variants</span><span class="o">.</span><span class="n">vcf</span> \
        <span class="o">--</span><span class="n">data</span><span class="o">-</span><span class="n">sources</span><span class="o">-</span><span class="n">path</span> <span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">local</span><span class="o">/</span><span class="n">bioinformatics</span><span class="o">/</span><span class="n">databases</span><span class="o">/</span><span class="n">funcotator_dataSources</span> \
        <span class="o">--</span><span class="n">output</span><span class="o">-</span><span class="n">file</span><span class="o">-</span><span class="nb">format</span> <span class="n">VCF</span> \
        <span class="o">--</span><span class="n">reference</span> <span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">local</span><span class="o">/</span><span class="n">bioinformatics</span><span class="o">/</span><span class="n">databases</span><span class="o">/</span><span class="n">Genome</span><span class="o">/</span><span class="n">Homo_sapiens_GRCh38</span><span class="o">.</span><span class="n">fasta</span> \
        <span class="o">--</span><span class="n">ref</span><span class="o">-</span><span class="n">version</span> <span class="n">hg38</span> \
        <span class="o">--</span><span class="n">output</span> <span class="n">S1_func_annotated_qd_dp_filtered_variants</span><span class="o">.</span><span class="n">vcf</span>
</pre></div>
</div>
<div class="admonition important">
<p class="admonition-title">Importante</p>
<p>Para descargar fuentes de anotación, es necesario entrar al servidor <a class="reference external" href="ftp://gsapubftp-anonymous&#64;ftp.broadinstitute.org/bundle/">ftp</a> del Broad Institute con nombre de usuario <code class="code docutils literal notranslate"><span class="pre">gsapubftp-anonymous</span></code> y contraseña <code class="code docutils literal notranslate"><span class="pre">gsapubftp-anonymous</span></code></p>
</div>
</div></blockquote>
</li>
</ol>
</div>
<div class="section" id="formatos-formatos-everywhere">
<h1>Formatos… formatos everywhere<a class="headerlink" href="#formatos-formatos-everywhere" title="Enlazar permanentemente con este título">¶</a></h1>
<div class="toggle admonition">
<p class="admonition-title">BAM</p>
<p>El formato BAM es similar en estructura al formato SAM, sin embargo, es el formato de elección para el manejo de alineamientos ya que está comprimido y ahorra mucho espacio</p>
</div>
<div class="toggle admonition">
<p class="admonition-title">BED</p>
<p>El formato BED consta de archivos de texto plano con estructura tabular que indica posiciones dentro de un genoma</p>
<p>Este formato es indispensable para el manejo óptimo de alineamientos ya que nos permite filtrar unicamente por las regiones que nos interesan!</p>
<p>Hay distintas versiones del formato <a class="reference external" href="https://genome.ucsc.edu/FAQ/FAQformat.html#format1">BED</a>, la estructura más básica consta de 4 a 6 columnas:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mi">13</span>      <span class="mi">32316459</span>        <span class="mi">32316526</span>        <span class="n">BRCA2</span>   <span class="o">.</span>       <span class="o">+</span>
<span class="mi">13</span>      <span class="mi">32319075</span>        <span class="mi">32319324</span>        <span class="n">BRCA2</span>   <span class="o">.</span>       <span class="o">+</span>
<span class="mi">13</span>      <span class="mi">32325074</span>        <span class="mi">32325183</span>        <span class="n">BRCA2</span>   <span class="o">.</span>       <span class="o">+</span>
</pre></div>
</div>
<div class="admonition important">
<p class="admonition-title">Importante</p>
<p>El formato BED es 0-based para la segunda columna!</p>
<p>Si mi gen inicia en el nucleótido 345 y termina en el 678 del cromosoma MT, su nomenclatura en formato BED será la siguiente:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">MT</span>      <span class="mi">344</span>     <span class="mi">678</span>     <span class="n">mi_gen</span>  <span class="o">.</span>       <span class="o">+</span>
</pre></div>
</div>
</div>
</div>
<div class="toggle admonition">
<p class="admonition-title">VCF</p>
<p>El formato VCF también es un archivo de texto plano con una estructura tabular:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1">#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  g204</span>
<span class="mi">1</span>       <span class="mi">13417</span>   <span class="n">rs777038595</span>     <span class="n">C</span>       <span class="n">CGAGA</span>   <span class="mf">324.73</span>  <span class="n">PASS</span>    <span class="n">AC</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">AF</span><span class="o">=</span><span class="mf">0.500</span><span class="p">;</span><span class="n">AN</span><span class="o">=</span><span class="mi">2</span><span class="p">;</span><span class="n">BaseQRankSum</span><span class="o">=</span><span class="mf">0.583</span><span class="p">;</span><span class="n">DB</span><span class="p">;</span><span class="n">DP</span><span class="o">=</span><span class="mi">26</span><span class="p">;</span><span class="n">ExcessHet</span><span class="o">=</span><span class="mf">3.0103</span><span class="p">;</span><span class="n">FS</span><span class="o">=</span><span class="mf">0.000</span><span class="p">;</span><span class="n">MLEAC</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">MLEAF</span><span class="o">=</span><span class="mf">0.500</span><span class="p">;</span><span class="n">MQ</span><span class="o">=</span><span class="mf">22.21</span><span class="p">;</span><span class="n">MQRankSum</span><span class="o">=-</span><span class="mf">0.032</span><span class="p">;</span><span class="n">QD</span><span class="o">=</span><span class="mf">12.49</span><span class="p">;</span><span class="n">ReadPosRankSum</span><span class="o">=-</span><span class="mf">2.185</span><span class="p">;</span><span class="n">SOR</span><span class="o">=</span><span class="mf">0.446</span><span class="p">;</span><span class="n">VQSLOD</span><span class="o">=</span><span class="mf">0.698</span><span class="p">;</span><span class="n">culprit</span><span class="o">=</span><span class="n">ReadPosRankSum</span> <span class="n">GT</span><span class="p">:</span><span class="n">AD</span><span class="p">:</span><span class="n">DP</span><span class="p">:</span><span class="n">GQ</span><span class="p">:</span><span class="n">PL</span>  <span class="mi">0</span><span class="o">/</span><span class="mi">1</span><span class="p">:</span><span class="mi">15</span><span class="p">,</span><span class="mi">11</span><span class="p">:</span><span class="mi">26</span><span class="p">:</span><span class="mi">99</span><span class="p">:</span><span class="mi">362</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">634</span>
</pre></div>
</div>
<div class="toggle admonition">
<p class="admonition-title">Descripción extendida del formato vcf</p>
<blockquote>
<div><table class="docutils align-default">
<colgroup>
<col style="width: 4%" />
<col style="width: 11%" />
<col style="width: 85%" />
</colgroup>
<tbody>
<tr class="row-odd"><td rowspan="2"><p>1: CHROM</p></td>
<td rowspan="2"><p>1</p></td>
<td rowspan="2"><p>Cromosoma en dónde se ubica la variante</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>2: POS</p></td>
<td rowspan="2"><p>1053827</p></td>
<td rowspan="2"><p>Posición de la variante en el cromosoma</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>3: ID</p></td>
<td rowspan="2"><p>rs74685771</p></td>
<td rowspan="2"><p>Identificador de la variante</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>4: REF</p></td>
<td rowspan="2"><p>G</p></td>
<td rowspan="2"><p>Alelo de referencia</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>5: ALT</p></td>
<td rowspan="2"><p>C</p></td>
<td rowspan="2"><p>Alelo encontrado en la muestra</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>6: QUAL</p></td>
<td rowspan="2"><p>856.77</p></td>
<td rowspan="2"><p>Score de calidad de la variante</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>7: FILTER</p></td>
<td rowspan="2"><p>PASS</p></td>
<td rowspan="2"><p>Flag de calidad de la variante</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="34"><p>8:INFO</p></td>
<td rowspan="2"><p>AC=1;</p></td>
<td rowspan="2"><p>Número de alelos alternos encontrados</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>AF=0.500;</p></td>
<td rowspan="2"><p>Frecuencia alelica (alelo alterno)</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>AN=2;</p></td>
<td rowspan="2"><p>Número total de alelos para una variante</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>BaseQRankSum=-2.129</p></td>
<td rowspan="2"><p>Z-score de la calidad de las bases mapeadas en el alelo alterno vs el alelo de referencia empleando una prueba de Wilcoxon</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>ClippingRankSum=-0.286;</p></td>
<td rowspan="2"><p>Z-score del número de lecturas con calidad de mapeo bajas para el alelo alterno vs el alelo de referencia empleando una prueba de Wilcoxon</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>DB;</p></td>
<td rowspan="2"><p>Indica si la variante pertenece a una base de datos (dbSNP)</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>DP=63;</p></td>
<td rowspan="2"><p>número de lecturas mapeadas en la posición de la variante</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>FS=0;</p></td>
<td rowspan="2"><p>P-value ajustado para determinar sesgo hacía una cadena de DNA empleando una prueba exacta de Fisher</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>MLEAC=1;</p></td>
<td rowspan="2"><p>Máxima verosimilitud esperada para el número de alelos observados (depende del número de muestras).</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>MLEAF=0.500;</p></td>
<td rowspan="2"><p>Máxima verosimilitud esperada para las frecuencias alélicas (depende del número de muestras).</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>MQ=60.00;</p></td>
<td rowspan="2"><p>Calidad media de mapeo en la posición de la variante.</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>MQRankSum=-0.0.635;</p></td>
<td rowspan="2"><p>Z-score de la calidad de mapeo las bases alíneadas en el alelo alterno vs el alelo de referencia empleando una prueba de Wilcoxon</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>QD=13.60;</p></td>
<td rowspan="2"><p>Confianza del alelo observado (Quality over depth)</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>ReadPosRankSum=0.621;</p></td>
<td rowspan="2"><p>Z-score del sesgo posicional (con respecto de la longitud de la lectura) del alelo alternativo vs el alelo de referencia (depende del número de muestras)</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>SOR=0.768;</p></td>
<td rowspan="2"><p>Suma simétrica de la razón de momios del número de lecturas que presentan la variante cerca del final de la lectura, vs las lecturas que la presentan cerca del inicio de la lectura.</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>VQSLOD=3.39;</p></td>
<td rowspan="2"><p>Logaritmo de la razón de momios de que la variante sea verdadera vs que sea un falso positivo.</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>culprit=qd</p></td>
<td rowspan="2"><p>El descriptor que presentó la métrica más desfavorable</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="10"><p>9:FORMAT</p></td>
<td rowspan="2"><p>GT</p></td>
<td rowspan="2"><p>Genotipo</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>AD</p></td>
<td rowspan="2"><p>Profundidad de los alelos observados (Ref,Alt)</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>DP</p></td>
<td rowspan="2"><p>Profundidad total (número de lecturas</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>GQ</p></td>
<td rowspan="2"><p>Calidad del genotipado</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>PL</p></td>
<td rowspan="2"><p>Verosimilitud de genotipos (Obs,Nul,Alt)</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="10"><p>10:g204</p></td>
<td rowspan="2"><p>0/1</p></td>
<td rowspan="2"><p>Heterocigoto</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>37,26</p></td>
<td rowspan="2"><p>(G)37 reads + (C)26 reads</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>63</p></td>
<td rowspan="2"></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>99</p></td>
<td rowspan="2"></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>885,0,1386</p></td>
<td rowspan="2"></td>
</tr>
<tr class="row-even"></tr>
</tbody>
</table>
</div></blockquote>
</div>
<div class="admonition important">
<p class="admonition-title">Importante</p>
<p>El formato VCF es 1-based en la segunda columna, por lo que es importante considerar esto cuando manejemos combinaciones de formatos (como con bedtools y bcftools)</p>
</div>
</div>
<div class="toggle admonition">
<p class="admonition-title">GFF</p>
<p>El formato GFF consta de archivos de texto plano con estructura tabular que indica posiciones dentro de un genoma, pueden ser genes, mRNAs, exones, regiones codificantes, y un largo etcetera.</p>
<p>Este formato es uno de los estándares de anotación genómica y se emplea en todos los proyectos de genómica como referencia para obtener genes</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mi">1</span>       <span class="n">havana</span>  <span class="n">exon</span>    <span class="mi">11869</span>   <span class="mi">12227</span>   <span class="o">.</span>       <span class="o">+</span>       <span class="o">.</span>       <span class="n">Parent</span><span class="o">=</span><span class="n">transcript</span><span class="p">:</span><span class="n">ENST00000456328</span><span class="p">;</span><span class="n">Name</span><span class="o">=</span><span class="n">ENSE00002234944</span><span class="p">;</span><span class="n">constitutive</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">ensembl_end_phase</span><span class="o">=-</span><span class="mi">1</span><span class="p">;</span><span class="n">ensembl_phase</span><span class="o">=-</span><span class="mi">1</span><span class="p">;</span><span class="n">exon_id</span><span class="o">=</span><span class="n">ENSE00002234944</span><span class="p">;</span><span class="n">rank</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">version</span><span class="o">=</span><span class="mi">1</span>
<span class="mi">1</span>       <span class="n">havana</span>  <span class="n">lnc_RNA</span> <span class="mi">11869</span>   <span class="mi">14409</span>   <span class="o">.</span>       <span class="o">+</span>       <span class="o">.</span>       <span class="n">ID</span><span class="o">=</span><span class="n">transcript</span><span class="p">:</span><span class="n">ENST00000456328</span><span class="p">;</span><span class="n">Parent</span><span class="o">=</span><span class="n">gene</span><span class="p">:</span><span class="n">ENSG00000223972</span><span class="p">;</span><span class="n">Name</span><span class="o">=</span><span class="n">DDX11L1</span><span class="o">-</span><span class="mi">202</span><span class="p">;</span><span class="n">biotype</span><span class="o">=</span><span class="n">processed_transcript</span><span class="p">;</span><span class="n">tag</span><span class="o">=</span><span class="n">basic</span><span class="p">;</span><span class="n">transcript_id</span><span class="o">=</span><span class="n">ENST00000456328</span><span class="p">;</span><span class="n">transcript_support_level</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">version</span><span class="o">=</span><span class="mi">2</span>
<span class="mi">1</span>       <span class="n">havana</span>  <span class="n">pseudogene</span>      <span class="mi">11869</span>   <span class="mi">14409</span>   <span class="o">.</span>       <span class="o">+</span>       <span class="o">.</span>       <span class="n">ID</span><span class="o">=</span><span class="n">gene</span><span class="p">:</span><span class="n">ENSG00000223972</span><span class="p">;</span><span class="n">Name</span><span class="o">=</span><span class="n">DDX11L1</span><span class="p">;</span><span class="n">biotype</span><span class="o">=</span><span class="n">transcribed_unprocessed_pseudogene</span><span class="p">;</span><span class="n">description</span><span class="o">=</span><span class="n">DEAD</span><span class="o">/</span><span class="n">H</span><span class="o">-</span><span class="n">box</span> <span class="n">helicase</span> <span class="mi">11</span> <span class="n">like</span> <span class="mi">1</span> <span class="p">[</span><span class="n">Source</span><span class="p">:</span><span class="n">HGNC</span> <span class="n">Symbol</span><span class="o">%</span><span class="mi">3</span><span class="n">BAcc</span><span class="p">:</span><span class="n">HGNC</span><span class="p">:</span><span class="mi">37102</span><span class="p">];</span><span class="n">gene_id</span><span class="o">=</span><span class="n">ENSG00000223972</span><span class="p">;</span><span class="n">logic_name</span><span class="o">=</span><span class="n">havana_homo_sapiens</span><span class="p">;</span><span class="n">version</span><span class="o">=</span><span class="mi">5</span>
<span class="mi">1</span>       <span class="n">havana</span>  <span class="n">exon</span>    <span class="mi">12010</span>   <span class="mi">12057</span>   <span class="o">.</span>       <span class="o">+</span>       <span class="o">.</span>       <span class="n">Parent</span><span class="o">=</span><span class="n">transcript</span><span class="p">:</span><span class="n">ENST00000450305</span><span class="p">;</span><span class="n">Name</span><span class="o">=</span><span class="n">ENSE00001948541</span><span class="p">;</span><span class="n">constitutive</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">ensembl_end_phase</span><span class="o">=-</span><span class="mi">1</span><span class="p">;</span><span class="n">ensembl_phase</span><span class="o">=-</span><span class="mi">1</span><span class="p">;</span><span class="n">exon_id</span><span class="o">=</span><span class="n">ENSE00001948541</span><span class="p">;</span><span class="n">rank</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">version</span><span class="o">=</span><span class="mi">1</span>
<span class="mi">1</span>       <span class="n">havana</span>  <span class="n">pseudogenic_transcript</span>  <span class="mi">12010</span>   <span class="mi">13670</span>   <span class="o">.</span>       <span class="o">+</span>       <span class="o">.</span>       <span class="n">ID</span><span class="o">=</span><span class="n">transcript</span><span class="p">:</span><span class="n">ENST00000450305</span><span class="p">;</span><span class="n">Parent</span><span class="o">=</span><span class="n">gene</span><span class="p">:</span><span class="n">ENSG00000223972</span><span class="p">;</span><span class="n">Name</span><span class="o">=</span><span class="n">DDX11L1</span><span class="o">-</span><span class="mi">201</span><span class="p">;</span><span class="n">biotype</span><span class="o">=</span><span class="n">transcribed_unprocessed_pseudogene</span><span class="p">;</span><span class="n">tag</span><span class="o">=</span><span class="n">basic</span><span class="p">;</span><span class="n">transcript_id</span><span class="o">=</span><span class="n">ENST00000450305</span><span class="p">;</span><span class="n">transcript_support_level</span><span class="o">=</span><span class="n">NA</span><span class="p">;</span><span class="n">version</span><span class="o">=</span><span class="mi">2</span>
<span class="mi">1</span>       <span class="n">havana</span>  <span class="n">exon</span>    <span class="mi">12179</span>   <span class="mi">12227</span>   <span class="o">.</span>       <span class="o">+</span>       <span class="o">.</span>       <span class="n">Parent</span><span class="o">=</span><span class="n">transcript</span><span class="p">:</span><span class="n">ENST00000450305</span><span class="p">;</span><span class="n">Name</span><span class="o">=</span><span class="n">ENSE00001671638</span><span class="p">;</span><span class="n">constitutive</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">ensembl_end_phase</span><span class="o">=-</span><span class="mi">1</span><span class="p">;</span><span class="n">ensembl_phase</span><span class="o">=-</span><span class="mi">1</span><span class="p">;</span><span class="n">exon_id</span><span class="o">=</span><span class="n">ENSE00001671638</span><span class="p">;</span><span class="n">rank</span><span class="o">=</span><span class="mi">2</span><span class="p">;</span><span class="n">version</span><span class="o">=</span><span class="mi">2</span>
<span class="mi">1</span>       <span class="n">havana</span>  <span class="n">exon</span>    <span class="mi">12613</span>   <span class="mi">12697</span>   <span class="o">.</span>       <span class="o">+</span>       <span class="o">.</span>       <span class="n">Parent</span><span class="o">=</span><span class="n">transcript</span><span class="p">:</span><span class="n">ENST00000450305</span><span class="p">;</span><span class="n">Name</span><span class="o">=</span><span class="n">ENSE00001758273</span><span class="p">;</span><span class="n">constitutive</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">ensembl_end_phase</span><span class="o">=-</span><span class="mi">1</span><span class="p">;</span><span class="n">ensembl_phase</span><span class="o">=-</span><span class="mi">1</span><span class="p">;</span><span class="n">exon_id</span><span class="o">=</span><span class="n">ENSE00001758273</span><span class="p">;</span><span class="n">rank</span><span class="o">=</span><span class="mi">3</span><span class="p">;</span><span class="n">version</span><span class="o">=</span><span class="mi">2</span>
<span class="mi">1</span>       <span class="n">havana</span>  <span class="n">exon</span>    <span class="mi">12613</span>   <span class="mi">12721</span>   <span class="o">.</span>       <span class="o">+</span>       <span class="o">.</span>       <span class="n">Parent</span><span class="o">=</span><span class="n">transcript</span><span class="p">:</span><span class="n">ENST00000456328</span><span class="p">;</span><span class="n">Name</span><span class="o">=</span><span class="n">ENSE00003582793</span><span class="p">;</span><span class="n">constitutive</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">ensembl_end_phase</span><span class="o">=-</span><span class="mi">1</span><span class="p">;</span><span class="n">ensembl_phase</span><span class="o">=-</span><span class="mi">1</span><span class="p">;</span><span class="n">exon_id</span><span class="o">=</span><span class="n">ENSE00003582793</span><span class="p">;</span><span class="n">rank</span><span class="o">=</span><span class="mi">2</span><span class="p">;</span><span class="n">version</span><span class="o">=</span><span class="mi">1</span>
<span class="mi">1</span>       <span class="n">havana</span>  <span class="n">exon</span>    <span class="mi">12975</span>   <span class="mi">13052</span>   <span class="o">.</span>       <span class="o">+</span>       <span class="o">.</span>       <span class="n">Parent</span><span class="o">=</span><span class="n">transcript</span><span class="p">:</span><span class="n">ENST00000450305</span><span class="p">;</span><span class="n">Name</span><span class="o">=</span><span class="n">ENSE00001799933</span><span class="p">;</span><span class="n">constitutive</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">ensembl_end_phase</span><span class="o">=-</span><span class="mi">1</span><span class="p">;</span><span class="n">ensembl_phase</span><span class="o">=-</span><span class="mi">1</span><span class="p">;</span><span class="n">exon_id</span><span class="o">=</span><span class="n">ENSE00001799933</span><span class="p">;</span><span class="n">rank</span><span class="o">=</span><span class="mi">4</span><span class="p">;</span><span class="n">version</span><span class="o">=</span><span class="mi">2</span>
<span class="mi">1</span>       <span class="n">havana</span>  <span class="n">exon</span>    <span class="mi">13221</span>   <span class="mi">13374</span>   <span class="o">.</span>       <span class="o">+</span>       <span class="o">.</span>       <span class="n">Parent</span><span class="o">=</span><span class="n">transcript</span><span class="p">:</span><span class="n">ENST00000450305</span><span class="p">;</span><span class="n">Name</span><span class="o">=</span><span class="n">ENSE00001746346</span><span class="p">;</span><span class="n">constitutive</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">ensembl_end_phase</span><span class="o">=-</span><span class="mi">1</span><span class="p">;</span><span class="n">ensembl_phase</span><span class="o">=-</span><span class="mi">1</span><span class="p">;</span><span class="n">exon_id</span><span class="o">=</span><span class="n">ENSE00001746346</span><span class="p">;</span><span class="n">rank</span><span class="o">=</span><span class="mi">5</span><span class="p">;</span><span class="n">version</span><span class="o">=</span><span class="mi">2</span>
</pre></div>
</div>
<p>La estructura del formato GFF consta de 9 columnas:</p>
<blockquote>
<div><table class="docutils align-default">
<colgroup>
<col style="width: 26%" />
<col style="width: 74%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head" rowspan="2"><p>Campo</p></th>
<th class="head" rowspan="2"><p>Ejemplo</p></th>
</tr>
<tr class="row-even"></tr>
</thead>
<tbody>
<tr class="row-odd"><td rowspan="2"><p>1: CHROM</p></td>
<td rowspan="2"><p>1</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>2: SOURCE</p></td>
<td rowspan="2"><p>havana</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>3: FEATURE</p></td>
<td rowspan="2"><p>exon</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>4: START</p></td>
<td rowspan="2"><p>13221</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>5: END</p></td>
<td rowspan="2"><p>13374</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>6: SCORE</p></td>
<td rowspan="2"><p>.</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>7: STRAND</p></td>
<td rowspan="2"><p>+</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>8:FRAME</p></td>
<td rowspan="2"><p>.</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="16"><p>9:ATTRIBUTE</p></td>
<td rowspan="2"><p>Parent=transcript:ENST00000450305;</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>Name=ENSE00001746346;</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>constitutive=0;</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>ensembl_end_phase=-1;</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>ensembl_phase=-1;</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>exon_id=ENSE00001746346;</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>rank=5;</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>version=2</p></td>
</tr>
<tr class="row-even"></tr>
</tbody>
</table>
</div></blockquote>
<div class="admonition important">
<p class="admonition-title">Importante</p>
<p>El formato GFF es 1-based para la quinta columna!</p>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2021, @atgenomics, @vflorelo, @dianolasa, @zorbax

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
  
 <script type="text/javascript">
    $(document).ready(function() {
        $(".toggle > *").hide();
        $(".toggle .admonition-title").show();
        $(".toggle .admonition-title").click(function() {
            $(this).parent().children().not(".admonition-title").toggle(400);
            $(this).parent().children(".admonition-title").toggleClass("open");
        })
    });
</script>


</body>
</html>